core/src/test/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorterSuite.java

```diff
@@ -101,7 +101,7 @@ public OutputStream apply(OutputStream stream) {
   public void setUp() {
     MockitoAnnotations.initMocks(this);
     sparkConf = new SparkConf();
-    tempDir = new File(Utils.createTempDir$default$1());
+    tempDir = Utils.createTempDir(System.getProperty("java.io.tmpdir"), "unsafe-test");
```

Comment:
This has to be deleted in `tearDown` too I presume in order to fix this? or else it's still reusing a same dir? 

I checked and there is a similar problem in some other Java tests:

```
JavaSaveLoadSuite
JavaMetastoreDataSourcesSuite
JavaAPISuite
```